### PR TITLE
[needinfo_regression_author] Ask the regression author to set the severity

### LINKED ISSUE #1518: Ask the author of the regression patch to set the severity instead of the triage owner
> _This issue is based on the autonag feedback survey._

In case of new regressions, when needinfoing the author of the patch causing the regression we should also ask them to set severity instead of needinfoing the triage owner too.

### PR DESCRIPTION
**Pull request by suhaibmujahid**
Resolves #1518 

## Checklist

- [x] Type annotations added to new functions
- [x] Docs added to functions touched in main classes
- [x] Dry-run produced the expected results
- [ ] The [`to-be-announced`](https://github.com/mozilla/relman-auto-nag/labels/to-be-announced) tag added if this is worth announcing

### CODE REVIEW

File: auto_nag/tests/test_user_activity.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        for email in self.disabled_users:
+            self.assertIn(email, users_info)
+
+            user = users_info[email]
+            self.assertIn("nick", user)
+            self.assertIn("id", user)
+            self.assertEqual(user["status"], UserStatus.DISABLED)
+
+        user = users_info["bdahl@mozilla.com"]
+        user["id"] = 425126
```
Comment (Archaeopteryx) [inline auto_nag/tests/test_user_activity.py:97]: Are this and the next line needed?

**Review by Archaeopteryx — APPROVED** [review]

File: auto_nag/tests/test_user_activity.py (hunk 1 above)
Comment (suhaibmujahid) [reply auto_nag/tests/test_user_activity.py]: Sorry, it is supposed to be an asserion. Fixed in 8cb5c80c6fc7ff434bfa8f8c3bb0735500d76550.

File: auto_nag/tests/test_user_activity.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+from auto_nag.people import People
+from auto_nag.user_activity import UserActivity, UserStatus
+
+
+class UserActivityTest(MockTestCase):
+    """Test the UserActivity class"""
+
+    mock_urls = [BugzillaUser.URL]
+
+    active_users = {"smujahid@mozilla.com", "mcastelluccio@mozilla.com"}
```
[the remark this answers is not in the document]
Comment (suhaibmujahid) [reply auto_nag/tests/test_user_activity.py:18]: I agree with you. But we need here to have emails linked to real accounts on Bugzilla so the API will return valid results (`responses` saves the results to mock future requests).

This approach is used in `libmozdata`: https://github.com/mozilla/libmozdata/blob/2d0d7731b4912f66987747c9725d25152f144989/tests/test_bugzilla.py#L89

**Comment by suhaibmujahid** [comment]
# Dry-run


## Example with the severity is not set (Bug [1786259](https://bugzilla.mozilla.org/show_bug.cgi?id=1786259))

:jfkthame, since you are the author of the regressor, bug 1756474, could you take a look? Also, could you set the severity field?

For more information, please visit [auto_nag documentation](https://wiki.mozilla.org/Release_Management/autonag#needinfo_regression_author.py).







## Example with the severity is set  (Bug [1786088](https://bugzilla.mozilla.org/show_bug.cgi?id=1786088))

:saschanaz, since you are the author of the regressor, bug 1785739, could you take a look?

For more information, please visit [auto_nag documentation](https://wiki.mozilla.org/Release_Management/autonag#needinfo_regression_author.py).

File: auto_nag/auto_mock.py [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+            )
+
+    def _request_callback(self, request):
+        logger.debug("Mock request %s %s", request.method, request.url)
+        path = self._build_path(request.method, request.url)
+
+        if os.path.exists(path):
+            # Load local file
+            logger.info("Using mock file %s", path)
+            with gzip.open(path, "rb") as file:
```
Comment (marco-c) [inline auto_nag/auto_mock.py:52]: Maybe we should not gzip, to have only readable files in the repo and normal diffs.
There aren't too many mock files, so we are not gaining much in compressing them.

File: auto_nag/scripts/needinfo_regression_author.py [hunk 4]
```diff
@@ -143,6 +159,16 @@ def comment_handler(bug, bug_id):
 
         return bugs
 
+    @staticmethod
+    def _has_editbugs_group(user_info: dict) -> bool:
+        """Check if the user has the editbugs group permissions"""
+
+        # All employees has edit bugs permissions
```
Comment (marco-c) [inline auto_nag/scripts/needinfo_regression_author.py:166]:
```suggestion
        # All employees have edit bugs permissions
```

Comment (marco-c) [inline auto_nag/scripts/needinfo_regression_author.py:166]: Is it actually the case? Can you double check with dkl or glob?

Comment (suhaibmujahid) [reply auto_nag/scripts/needinfo_regression_author.py:166]: Yes, it is. I double-checked with dkl on #bmo: https://mozilla.slack.com/archives/C5F7KFJ4R/p1660769611872729

File: auto_nag/auto_mock.py (hunk 3 above)
Comment (suhaibmujahid) [reply auto_nag/auto_mock.py]: Done in 954ec05e0ba3cc94561822fcec883cc74ab3c105

**Review by marco-c — APPROVED** [review]

**Pull request merged**